wandb/sdk/artifacts/artifact.py

```diff
@@ -113,8 +114,22 @@ class Artifact:
         ```
     """
 
-    _TMP_DIR = tempfile.TemporaryDirectory("wandb-artifacts")
-    atexit.register(_TMP_DIR.cleanup)
+    _TMP_DIR: Optional[tempfile.TemporaryDirectory] = None
+    _TMP_DIR_LOCK = threading.Lock()
+
+    @classmethod
+    def _get_media_tmp_dir(cls) -> tempfile.TemporaryDirectory:
+        if cls._TMP_DIR:
+            return cls._TMP_DIR
+
+        with cls._TMP_DIR_LOCK:
+            # Check again in case another thread created it while we were waiting
+            if cls._TMP_DIR:
+                return cls._TMP_DIR
+
+            cls._TMP_DIR = tempfile.TemporaryDirectory("wandb-artifacts")
+            atexit.register(cls._TMP_DIR.cleanup)  # Register the cleanup only once
+            return cls._TMP_DIR
```

Comment:
Instead of lock usage for mulithreading, if your concern is to not create multiple tempdirs if a tempdir has already been created why not

```
try:
      cls._TMP_DIR = tempfile.TemporaryDirectory("wandb-artifacts")
      ... <if needed, throw exception on exists that contains existing dir> ...
except Exception as e:
     return e.get_existing_dir()
```

i might be missing context here on why this is not possible